libkbfs/disk_block_cache_wrapped.go

```diff
@@ -42,38 +42,59 @@ type diskBlockCacheWrapped struct {
 
 var _ DiskBlockCache = (*diskBlockCacheWrapped)(nil)
 
-func newDiskBlockCacheWrapped(config diskBlockCacheConfig, storageRoot string) (
-	cache *diskBlockCacheWrapped, err error) {
-	workingSetCacheRoot := filepath.Join(storageRoot, workingSetCacheFolderName)
-	workingSetCache, err := newDiskBlockCacheStandard(config,
-		workingSetCacheLimitTrackerType, workingSetCacheRoot)
-	if err != nil {
-		return nil, err
+func (cache *diskBlockCacheWrapped) enableCache(
+	typ diskLimitTrackerType, cacheFolder string) (err error) {
+	cache.mtx.Lock()
+	defer cache.mtx.Unlock()
+	var cachePtr *DiskBlockCache
+	switch typ {
+	case syncCacheLimitTrackerType:
+		cachePtr = &cache.syncCache
+	case workingSetCacheLimitTrackerType:
+		cachePtr = &cache.workingSetCache
+	default:
+		return errors.New("invalid disk cache type")
+	}
+	if *cachePtr != nil {
+		// We already have a cache of the desired type. Thus, this method is
+		// idempotent.
+		return nil
 	}
-	cache = &diskBlockCacheWrapped{
-		config:          config,
-		storageRoot:     storageRoot,
-		workingSetCache: workingSetCache,
-		syncCache:       nil,
+	if cache.storageRoot == "" {
```

Comment:
This is kind of a weird check -- could you use the test mode flag you introduced in this PR?  Or be even more explicit by having a `newDiskBlockCacheWrappedForTest` that passes some special flag into this function.